orangecontrib/text/widgets/utils/widgets.py

```diff
@@ -535,9 +535,10 @@ def __init__(self, widget, model_format, provider_format,
 
     def load_model(self, path_to_file):
         self.model_path = path_to_file
-        self.valueChanged.emit(self.model_path, self.resource_path)
+        if self.model_path:
+            self.valueChanged.emit(self.model_path, self.resource_path)
 
     def load_provider(self, path_to_file):
         self.resource_path = path_to_file
-        self.valueChanged.emit(self.model_path, self.resource_path)
-
+        if path_to_file and self.model_path:
+            self.valueChanged.emit(self.model_path, self.resource_path)
```

Comment:
Did you try conditionalizing the [open_file call in FileWidget](https://github.com/kernc/orange3-text/blob/75ecce39394f6055ab5cf05aa3e1c5af8526ba32/orangecontrib/text/widgets/utils/widgets.py#L296-L296) instead?

The "fail early" guide advises to fail (or prevent further action) as soon as the conditions exist to do so.
https://stackoverflow.com/questions/2807241/what-does-the-expression-fail-early-mean-and-when-would-you-want-to-do-so